lib/msal-common/src/server/ServerTokenRequestParameters.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { ICrypto } from "../crypto/ICrypto";
+import { ScopeSet } from "../auth/ScopeSet";
+import { TokenExchangeParameters } from "../request/TokenExchangeParameters";
+import { ProtocolUtils } from "../utils/ProtocolUtils";
+import { Constants, HEADER_NAMES, AADServerParamKeys } from "../utils/Constants";
+import { ServerRequestParameters } from "./ServerRequestParameters";
+
+export class ServerTokenRequestParameters extends ServerRequestParameters {
+
+    // Params
+    tokenRequest: TokenExchangeParameters;
+
+    constructor(clientId: string, tokenRequest: TokenExchangeParameters, redirectUri: string, cryptoImpl: ICrypto) {
+        super(clientId, redirectUri, cryptoImpl);
+        this.tokenRequest = tokenRequest;
+
+        this.scopes = new ScopeSet(this.tokenRequest && this.tokenRequest.scopes, this.clientId, false);
+
+        const randomGuid = this.cryptoObj.createNewGuid();
+        this.state = ProtocolUtils.setRequestState(this.tokenRequest && this.tokenRequest.userRequestState, randomGuid);
+
+        this.correlationId = this.tokenRequest.correlationId || this.cryptoObj.createNewGuid();
+    }
+
+    createRequestHeaders(): Map<string, string> {
+        const headers = new Map<string, string>();
+        headers.set(HEADER_NAMES.CONTENT_TYPE, Constants.URL_FORM_CONTENT_TYPE);
+        return headers;
+    }
+
+    async createRequestBody(): Promise<string> {
+        const paramString = await this.createParamString();
+        return paramString.join("&");
+    }
+
+    protected async createParamString(): Promise<Array<string>> {
```

Comment:
Why is this `async`? Seems synchronous.

Reply:
It is synchronous for `ServerTokenRequestParameters` but asynchronous for `ServerCodeRequestParameters`, and since they both extend the same base class, they are both synchronous. I suppose this doesn't need to be defined in the base class, but since both child classes use it I figured it would be better to define it there.

Reply:
Does making the base class async affect the performance of the `ServerTokenRequestParameters` which can be synchronous?

Reply:
made it synchronous